baseband/vlbi_base/base.py

```diff
@@ -129,15 +129,39 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
                  sample_rate=None):
 
         if frames_per_second is None and sample_rate is None:
-            frames_per_second = self._get_frame_rate(fh_raw, type(header0))
+            try:
+                frames_per_second = self._get_frame_rate(fh_raw,
+                                                         type(header0))
+            except Exception as exc:
+                exc.args += ("_get_frame_rate raised {e.__class__.__name__} "
+                             "error before reading one second of data.  File "
+                             "may have less than one second of data, in "
+                             "which case framerate must manually be passed "
+                             "to file opener.  File could also be "
+                             "corrupt.".format(e=exc),)
+                raise
```

Comment:
I'm not sure if this is IPython-specific, but I noticed that raising a new error under `except Exception:` truncates the original traceback of the EOFError, which I'd really want to have as an end-user.  I'm happy to change the text to something like the above, though.